### DISCUSSION TITLE
Returning HEADER data from a Query / Mutation resolver

### DISCUSSION THREAD

**Discussion by bo55vxr** [ChilliCream/hotchocolate#5280] [Q&A]
Is there a way to return values in the Response HEADER from the Query / Mutation resolver.

For example, we have a "login" GraphQL query, which returns a dataset in the normal response body, but I want to return a "session token" which is then used in any subsequent query / mutation requests.

I would rather return this token in the Header, rather than as an extension to the dataset in the response body.

**Comment by tobias-tengler** [comment]
You should be able to set headers through the ASP.NET Core `HttpContext`: https://chillicream.com/docs/hotchocolate/fetching-data/resolvers/#accessing-the-httpcontext

**Reply by bo55vxr** [reply]
Sorry, I should have mentioned that we have deployed as an Azure Function, not a ASP.Net Web App

**Reply by tobias-tengler** [reply]
Did you try it? I haven't worked with Azure Functions much, but from a quick online search, the `IHttpContextAccessor` should also work for Azure Functions.

**Reply by michaelstaib** [reply]
Depends on the project kind ... I think with the integrated approach yes ... should work, but out of process should not work.
